apps/web/src/api/hooks/useUploadToStorage.ts

```diff
@@ -0,0 +1,42 @@
+import axios from 'axios';
+import { useMutation, UseMutationOptions } from '@tanstack/react-query';
+import { IResponseError, MIME_TYPE_TO_FILE_EXTENSION, UploadTypesEnum } from '@novu/shared';
+
+import { useGetSignedUrl } from './useGetSignedUrl';
+
+export const useUploadToStorage = (
+  options: UseMutationOptions<string, IResponseError, { file: File; type: UploadTypesEnum }> = {}
+) => {
+  const { getSignedUrl } = useGetSignedUrl();
+  const { mutateAsync: uploadToStorage, ...mutationData } = useMutation(
+    async ({ file, type }) => {
+      const { signedUrl, path, additionalHeaders } = await getSignedUrl({
+        extension: MIME_TYPE_TO_FILE_EXTENSION[file.type],
+        type,
+      });
+      await axios.put(signedUrl, file, {
+        headers: {
+          'Content-Type': file.type,
+          ...additionalHeaders,
+        },
+        transformRequest: [
+          (data, headers) => {
+            if (headers) {
+              // eslint-disable-next-line
+              delete headers.Authorization;
+            }
+
+            return data;
+          },
+        ],
+      });
+
+      return path;
+    },
+    {
+      ...options,
+    }
+  );
+
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:**  add the `onError` method with the `errorMessage` utility‏

Reply:
on error will be defined in the places where the hook is used ;)